Issue 3551 transition to secondary

shane-tomlinson opened the pull request:
@lloyd or @seanmonstar - mind reviewing?

An ephemeral instance is set up at transition-to-secondary

I have added a new selenium test suite - "transition-to-secondary". This case was not covered by our selenium tests before.

fixes #3551

Commit:
Fix the "transition_to_primary" upgrade.

fixes #3551

Commit:
Add Selenium and Frontend tests for #3551

seanmonstar commented on resources/static/dialog/js/misc/state.js:
```diff
@@ -448,9 +448,15 @@ BrowserID.State = (function() {
+          }
+          else {
+            redirectToState("authenticate", addressInfo);
+          }
```
this much looks fine, but I really don't know what the automated tests do... :(

Commit:
Simplify and clarify the Selenium and front end tests.

* To Selenium, simplify the tests.
* To Selenium, add better commenting.
* To Frontend tests, clarify the transition_to_secondary test names to be self explanatory.

shane-tomlinson commented:
@seanmonstar - I have updated the Selenium and QUnit tests to contain more context. I have also simplified the Selenium tests. The basic Selenium test flow is:

1) Create an account with a secondary
2) Add a primary address
3) Disable primary support for address in 2
4) Load up dialog, try to sign in using address from 2. Ensure user has to enter Persona password and verify email
5) Verify email, make sure the user is signed in.
